pkg/clientaccess/token.go

```diff
@@ -275,8 +289,9 @@ func (i *Info) Get(path string) ([]byte, error) {
 	return get(p.String(), GetHTTPClient(i.CACerts, i.CertFile, i.KeyFile), i.Username, i.Password, i.Token())
 }
 
-// Put makes a request to a subpath of info's BaseURL
-func (i *Info) Put(path string, body []byte) error {
+// Put makes a request to a subpath of info's BaseURL, with a 10 second timeout
+func (i *Info) Put(path string, body []byte, option ...ClientOption) error {
```

Comment:
I would probably add `option ...ClientOption` to `Get()` as well for consistency's sake.

I would also probably pass the options all the way into `GetHTTPClient()`, rather than mutating the returned client here. Options are usually passed all the way into whatever factory function creates the client, so that the client is returned with the client properly configured from the get-go.

Reply:
Fixed in [`7a62656` (#9340)](https://github.com/k3s-io/k3s/pull/9340/commits/7a62656ec6c36b95b4151106420173f4194700e3)